src/shaders/_prelude_fog.fragment.glsl

```diff
@@ -49,6 +42,19 @@ float fog_opacity(vec3 pos) {
     return fog_opacity((length(pos) - u_fog_range.x) / (u_fog_range.y - u_fog_range.x));
 }
 
+// This function applies haze to an input color using an approximation of the following algorithm:
+//   1. convert `color` from sRGB to linear RGB
+//   2. add haze (presuming haze is in linear RGB)
+//   3. tone-map the output
+//   4. convert the result back to sRGB
+// The equation below is based on a curve fit of the above algorithm, with the additional approximation
+// during linear-srgb conversion that gamma=2, in order to avoid transcendental function evaluations
+// which don't affect the visual quality.
+vec3 haze_apply(vec3 color, vec3 haze) {
+    vec3 color2 = color * color;
+    return sqrt((color2 + haze) / (1.0 + color2 * color2 * haze));
```

Comment:
If you try hard enough, you can blow out the colors and exceed 1.0 in a way which you couldn't before. In practice, for reasonable amounts of haze, I don't think this is a problem. Still I might back off this power ever so slightly.

```suggestion
    return sqrt((color2 + haze) / (1.0 + color2 * color * haze));
```

color^3 seems like a reasonable compromise/safety factor/splitting the difference.

color^2:

<img width="574" alt="Screen Shot 2021-04-21 at 9 25 51 PM" src="https://user-images.githubusercontent.com/572717/115655438-2c0b9100-a2e8-11eb-8dd7-82e25982230e.png">

color^3:

<img width="558" alt="Screen Shot 2021-04-21 at 9 25 00 PM" src="https://user-images.githubusercontent.com/572717/115655452-32017200-a2e8-11eb-8af9-e1059ec59031.png">

color^4

<img width="572" alt="Screen Shot 2021-04-21 at 9 23 22 PM" src="https://user-images.githubusercontent.com/572717/115655464-388fe980-a2e8-11eb-9596-abb8544648b9.png">